File: Laravel/app/Services/Handlers/DirectoryHandler.php
<?php
/**
 * Created by PhpStorm.
 * User: tsao
 * Date: 2017/11/5
 * Time: 下午 11:33
 */

namespace App\Services\Handlers;


use App\Services\Configs\Candidate;

/**
 * 備份目錄
 * Class DirectoryHandler
 * @author steve.tsao
 * @package App\Services\Handlers
 */
class DirectoryHandler extends AbstractHandler
{
    /**
     * 處理檔案目錄備份
     * @author steve.tsao
     * @param Candidate $candidate 檔案資訊
     * @param string|null $target 檔案內容
     * @return string 檔案內容
     */
    public function PerForm(Candidate $candidate, $target): string
    {
        $fileName = $this->GenFileName($candidate->Name) . '.bak';

        $this->ConvertStringToFile($candidate->Config->Dir . $fileName, $target);

        return $target;
    }
}

File: NetCore/ConsoleOop/ConsoleOop/Handlers/DirectoryHandler.cs
using MyBackupCandidate;

namespace ConsoleOop.Handlers
{
    /// <summary>
    /// 備份目錄
    /// </summary>
    class DirectoryHandler : AbstractHandler
    {
        /// <summary>
        /// 處理檔案目錄備份
        /// </summary>
        /// <param name="candidate">檔案資訊</param>
        /// <param name="target">檔案</param>
        /// <returns>備份後檔案</returns>
        public override byte[] PerForm(Candidate candidate, byte[] target)
        {
            byte[] result = target;

            if (target != null)
            {
                result = this.CopyToDirectory(candidate, target);
            }

            return result;
        }

        /// <summary>
        /// 備份檔案到目錄
        /// </summary>
        /// <param name="candidate">檔案資訊</param>
        /// <param name="target">檔案</param>
        /// <returns>備份後檔案</returns>
        private byte[] CopyToDirectory(Candidate candidate, byte[] target)
        {
            this.ConvertByteArrayToFile(candidate.Config.Dir + this.GenFileName(candidate.Name) + ".bak", target);

            return target;
        }
    }
}
